articles/sql-or-nosql-when-to-choose-what/index.md

```diff
@@ -0,0 +1,123 @@
+NoSQL example context:
+```sql
+Database --> Collection(s) --> Objects(s)
+```
+If we refer back to SQL, the collection would be the table and the object will be the rows and columns. Just like SQL where a database has several tables, NoSQL creates multiple collections. In this case, there are no relations between the collections. Rows and columns will be the objects of the NoSQL.
+Like SQL, NoSQL has a unique id that uniquely identifies a specific object within a collection. Nested objects and array are supported where an object can be inside another object. If we were to represent the SQL student info database in NoSQL, this how it would look like:
+
+```json
+Student {
+        _id: Objectld('14s5gg98we9d’),
+        firstName: Alex,
+        lastName: Brooks,
+        username: alex09_b,
+        Courses: [{name: Web Dev}, {name: React js}]
+    }
+```
+Some of the common NoSQL databases include [MongoDB](https://www.mongodb.com/what-is-mongodb), [Apache Cassandra](https://cassandra.apache.org/doc/latest/getting_started/index.html), [Google Cloud BigTable](https://cloud.google.com/bigtable), [Apache HBase](https://hbase.apache.org/), and [Redis](https://redis.io/).
+
+### SQL or NoSQL (high-level differences)
+
+#### Relation
+
+SQL employs a kind of relationship that exists between two or more database tables. As we have explained above, NoSQL is non-relational. To get this point correct, think of two towns let's say town A and town B. Take town A, where the residents understand only one language. They use it to do all sorts of communication and interaction around them day in day out. To change the language of this town will bring a lot of confusion and disruptions among the residents. Take town B, the residents speak different languages. Everyone communicates and interacts with the world around them differently. There is no universal way to understand each other. When a new resident is introduced to this town, the language of this new resident will affect no one.
+
+Town A represents the relational SQL, and Town B represents the non-relational NoSQL. SQL is powerful and versatile. On the other hand, they are restrictive and structured with a predefined schema. If you choose to use them, your data must use and follow the same structure. And as in Town A, changing its structure would be difficult. It will disrupt your whole system. The language of SQL is tabular with rows and columns, you can only follow that structure.
+
+NoSQL on the hand, Just like town B, it has a dynamic unstructured schema that is flexible to any new resident. Its data can be stored in many ways. This data can be
```

Comment:
**NoSQL on the hand, Just like town B**

"NoSQL on the other hand, like town B" would sound better.